pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -424,13 +423,13 @@ func (m *ManagerImpl) Register(ctx context.Context, r *pluginapi.RegisterRequest
 	}
 	if !versionCompatible {
 		errorString := fmt.Sprintf(errUnsupportedVersion, r.Version, pluginapi.SupportedVersions)
-		klog.Infof("Bad registration request from device plugin with resource name %q: %s", r.ResourceName, errorString)
+		klog.InfoS("Bad registration request from device plugin with resource", "resourceName", r.ResourceName, "err", fmt.Errorf(errorString))
```

Comment:
```suggestion
        err := fmt.Errorf(errUnsupportedVersion, r.Version, pluginapi.SupportedVersions)
		klog.InfoS("Bad registration request from device plugin with resource", "resourceName", r.ResourceName, "err", err)
		return &pluginapi.Empty{}, err
```
Can we simplify and create the error before? 

Sorry for invalid suggestion, but GitHub doesn't allow to provide suggestions for lines that were not changed.